src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -1024,16 +1024,10 @@ bool ReplicatedMergeTreeQueue::shouldExecuteLogEntry(
 {
     /// If our entry produce part which is already covered by
     /// some other entry which is currently executing, then we can postpone this entry.
-    if (entry.type == LogEntry::MERGE_PARTS
```

Comment:
I just don't understand why this if exists. If we already executing some entry that will produce `new_part_name` (or part covering it) then we should postpone anything that will produce the same or covered part.